core/runtime/register_trt_op.cpp

```diff
@@ -10,80 +10,87 @@ namespace trtorch {
 namespace core {
 namespace runtime {
 
-// SM Compute capability <Compute Capability, Device Name> map
-const std::unordered_map<std::string, std::string>& get_dla_supported_SM() {
-  // Xavier SM Compute Capability
-  static std::unordered_map<std::string, std::string> dla_supported_SM = {{"7.2", "Xavier"}};
-  return dla_supported_SM;
-}
-
 // Checks if the context switch requred for device ID
 bool is_switch_required(const CudaDevice& curr_device, const CudaDevice& conf_device) {
   // If SM capability is not the same as configured then switch
   if ((curr_device.major != conf_device.major) || (curr_device.minor != conf_device.minor)) {
-    LOG_WARNING("Configured SM capability does not match with current device ID. Switching context");
+    LOG_WARNING(
+        "Configured SM capability " << conf_device.getSMCapability()
+                                    << " does not match with current device SM capability "
+                                    << curr_device.getSMCapability() << " (" << curr_device
+                                    << "). Switching device context");
     return true;
   }
 
   // GPU case
   if (conf_device.device_type == nvinfer1::DeviceType::kGPU) {
     if (curr_device.device_name != conf_device.device_name) {
       LOG_WARNING(
-          "TRTEngine compiled for " << conf_device.device_name << " but current CUDA device is "
-                                    << curr_device.device_name << ". Switching the device context");
+          "Program compiled for " << conf_device.device_name << " but current CUDA device is " << curr_device
+                                  << ". Attempting to switch device context for better compatibility");
       return true;
     }
   }
 
   if (curr_device.id != conf_device.id) {
     LOG_WARNING(
         "Configured Device ID: " << conf_device.id << " is different that current device ID: " << curr_device.id
-                                 << ". Switching context");
+                                 << ". Moving input tensors to device: " << conf_device.id);
     return true;
   }
 
   return false;
 }
 
-int select_cuda_device(const CudaDevice& conf_device) {
-  int device_id = 0;
-  auto dla_supported = get_dla_supported_SM();
+CudaDevice select_cuda_device(const CudaDevice& conf_device) {
+  int64_t device_id = -1;
+  auto dla_supported = get_dla_supported_SMs();
 
-  auto device_list = cuda_device_list.instance().get_devices();
+  auto device_list = get_available_device_list().get_devices();
+
+  CudaDevice new_target_device;
 
   for (auto device : device_list) {
-    auto compute_cap = std::to_string(device.second.major) + "." + std::to_string(device.second.minor);
+    auto compute_cap = device.second.getSMCapability();
```

Comment:
Is getSMCapability function consistent with other function naming guidelines such as select_cuda_device?

Reply:
In general I use camel case for methods and snake for functions, but use whatever feels most natural in that specific case.